### PR TITLE
Updated Legend to make it closeable on small screens

### LINKED ISSUE #2880: Can’t close legend on mobile view
image

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Issue

Closes #2880

## Description

This ensures Legend stays on top of the other controls, so it can be closed on smaller screens.

Also cleaned up the components and removes all `scss` styling and replaced it with `styled-components`.

### CODE REVIEW

File: web/src/layout/legend.js [hunk 1]
```diff
@@ -11,8 +13,80 @@ import { solarColor, windColor } from '../helpers/scales';
...
+  <StyledMobileHeader collapsed={!isOpen}>
+    <span>{__('misc.legend')}</span>
+    <ToggleLegendButton className="material-icons" onClick={onToggle}>
+      {isOpen ? 'call_received' : 'call_made'}
+    </ToggleLegendButton>
+  </StyledMobileHeader>
+);
+
+const LegendItem = ({
+  enabled, label, unit, children,
```
Comment (skovhus) [inline web/src/layout/legend.js:80]: Suggestion: `enabled` -> `isEnabled`

**Review by skovhus — APPROVED** [review]
One suggest. But really nice that you also migrated to styled components.

File: web/src/layout/legend.js [hunk 2]
```diff
@@ -11,8 +13,80 @@ import { solarColor, windColor } from '../helpers/scales';
...
+  padding: 12px 10px 10px 10px;
+  margin: 16px;
+  box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
+  z-index: 99;
+  transition: width 0.4s, height 0.4s;
+  font-family: 'Euclid Triangle', 'Open Sans', sans-serif;
+  font-size: 0.8rem;
+  
+  // Apply specific styles if the legend is collapsed
+  ${props => props.collapsed
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:32]:
```suggestion
  ${props => props.isCollapsed
```

File: web/src/layout/legend.js [hunk 3]
```diff
@@ -11,8 +13,80 @@ import { solarColor, windColor } from '../helpers/scales';
...
+  position: absolute;
+  right: 5px;
+  top: 5px;
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+const StyledMobileHeader = styled.div`
+  text-align: ${props => (props.collapsed ? 'left' : 'center')};
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:60]:
```suggestion
  text-align: ${props => (props.isCollapsed ? 'left' : 'center')};
```

File: web/src/layout/legend.js [hunk 4]
```diff
@@ -11,8 +13,80 @@ import { solarColor, windColor } from '../helpers/scales';
...
+  margin-bottom: 5px;
+
+  @media (min-width: 768px) {
+    display: none;
+    text-align: center;
+  }
+}`;
+
+const MobileHeader = ({ onToggle, isOpen }) => (
+  <StyledMobileHeader collapsed={!isOpen}>
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:71]:
```suggestion
  <StyledMobileHeader isCollapsed={!isOpen}>
```

File: web/src/layout/legend.js (hunk 1 above)
Comment (madsnedergaard) [inline web/src/layout/legend.js:80]:
```suggestion
  isEnabled, label, unit, children,
```

File: web/src/layout/legend.js [hunk 5]
```diff
@@ -11,8 +13,80 @@ import { solarColor, windColor } from '../helpers/scales';
...
+    <span>{__('misc.legend')}</span>
+    <ToggleLegendButton className="material-icons" onClick={onToggle}>
+      {isOpen ? 'call_received' : 'call_made'}
+    </ToggleLegendButton>
+  </StyledMobileHeader>
+);
+
+const LegendItem = ({
+  enabled, label, unit, children,
+}) => (!enabled ? null : (
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:81]:
```suggestion
}) => (!isEnabled ? null : (
```

File: web/src/layout/legend.js [hunk 6]
```diff
@@ -31,64 +105,53 @@ const Legend = ({
...
 
   return (
-    <div className={`floating-legend-container ${mobileCollapsedClass}`}>
-      <div className="floating-legend-mobile-header">
-        <span>{__('misc.legend')}</span>
-        <i className="material-icons toggle-legend-button" onClick={toggleLegend}>
-          {legendVisible ? 'call_received' : 'call_made'}
-        </i>
-      </div>
+    <LegendsContainer collapsed={!legendVisible}>
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:113]:
```suggestion
    <LegendsContainer isCollapsed={!legendVisible}>
```

File: web/src/layout/legend.js [hunk 7]
```diff
@@ -31,64 +105,53 @@ const Legend = ({
...
-                markerColor="red"
-                ticksCount={5}
-              />
-            </div>
-          )}
-          <div className={`co2-legend floating-legend ${mobileCollapsedClass}`}>
-            <div className="legend-header">
-              {__('legends.carbonintensity')} <small>(gCO₂eq/kWh)</small>
-            </div>
+          <LegendItem label={__('legends.windpotential')} unit="m/s" enabled={windEnabled}>
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:117]:
```suggestion
          <LegendItem label={__('legends.windpotential')} unit="m/s" isEnabled={windEnabled}>
```

File: web/src/layout/legend.js [hunk 8]
```diff
@@ -31,64 +105,53 @@ const Legend = ({
...
+            />
+          </LegendItem>
+          <LegendItem
+            label={__('legends.solarpotential')}
+            unit={(
+              <span>
+                W/m<span className="sup">2</span>
+              </span>
+            )}
+            enabled={solarEnabled}
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:133]:
```suggestion
            isEnabled={solarEnabled}
```

File: web/src/layout/legend.js [hunk 9]
```diff
@@ -31,64 +105,53 @@ const Legend = ({
...
             <HorizontalColorbar
-              id="carbon-intensity-bar"
+              id="solar-potential-bar"
+              colorScale={solarColor}
+              currentValue={solarColorbarValue}
+              markerColor="red"
+              ticksCount={5}
+            />
+          </LegendItem>
+          <LegendItem label={__('legends.carbonintensity')} unit="gCO₂eq/kWh" enabled>
```
Comment (madsnedergaard) [inline web/src/layout/legend.js:143]:
```suggestion
          <LegendItem label={__('legends.carbonintensity')} unit="gCO₂eq/kWh" isEnabled>
```

**Review by madsnedergaard — COMMENTED** [review]
Renaming boolean props...

File: web/src/layout/legend.js (hunk 1 above)
Comment (madsnedergaard) [reply web/src/layout/legend.js]: Good point, fixed.

**Pull request merged**